lxd/images.go

```diff
@@ -353,18 +381,15 @@ func imageDelete(d *Daemon, r *http.Request) Response {
 	return EmptySyncResponse
 }
 
-func imageGet(d *Daemon, r *http.Request) Response {
-	fingerprint := mux.Vars(r)["fingerprint"]
-
-	public := !d.isTrustedClient(r)
+func doImageGet(d *Daemon, fingerprint string, public bool) (shared.ImageInfo, Response) {
 	imgInfo, err := dbImageGet(d, fingerprint, public)
 	if err != nil {
-		return SmartError(err)
+		return shared.ImageInfo{}, SmartError(err)
```

Comment:
Can you pass `nil` in all of these? I think that's probably more idiomatic than the zero struct.